content/deyemiobaa/building-a-todolist-with-rails.md

```diff
@@ -0,0 +1,351 @@
+
+```html
+<h1>Hello World</h1>
+```
+
+Then add the following code to the `config/routes.rb` file.
+
+```ruby
+Rails.application.routes.draw do
+  get '/todos', to: 'todos#index'
+end
+```
+
+This will create a route that maps the `/todos` url to the `todos#index` controller method. The `get` method specifies the HTTP verb to use. The `to` option specifies the controller and method to use. The `todos#index` syntax is a shorthand for `controller: 'todos', action: 'index'`.
+
+Now run `rails s` or `bin/dev` to start the server, and visit `http://localhost:3000/todos` in your browser. You should see the 'Hello World' message.
+
+## Using ERB to write logic and display content in HTML.
+
+ERB stands for Embedded Ruby. It is a templating system that allows us to write ruby code within HTML files. This is useful because it allows us to write logic and display content in HTML. We can also use it to write HTML in a more concise way. 
+This aspect of Rails is the V in MVC. The view layer is responsible for displaying the content to the user. It is the part of the application that handles the presentation logic. The view layer interacts with the controller to get the data it needs to display. Without controllers, views would not be able to display any data.
+
+On our webpage, we would have a simple form for creating new todo items, and a table to list all the todo items. We'll use ERB to write the logic for this.
+
+Open the `app/views/todos/index.html.erb` file and add the following code.
+
+```erb
+<h1 class="text-2xl">Todo List</h1>
+
+<%= form_with(model: @todo, class: "my-10") do |form| %>
+  <div class="my-5">
+    <%= form.label :description %>
+    <%= form.text_field :description, placeholder: "new todo", class: "block shadow rounded-md border border-gray-200 outline-none px-3 py-2 mt-2 w-2/4" %>
+  </div>
+
+  <div class="inline">
+    <%= form.submit class: "rounded-lg py-3 px-5 bg-blue-600 text-white inline-block font-medium cursor-pointer" %>
+  </div>
+<% end %>
+```
+
+This code will render a form with a text field and a submit button. The `form_with` method is a helper method that creates a form for a given model. The `model` option specifies the model to use. The `class` option specifies the css class to use for the form. The `do |form|` block specifies the content of the form. The `form.label` method creates a label for the text field. The `form.text_field` method creates a text field for the form. The `form.submit` method creates a submit button for the form.
+
+To view this on our webpage, we need to create a new instance of the Todo model in the controller. Open the `app/controllers/todos_controller.rb` file and replace what you have with the following code.
+
+```ruby
+class TodosController < ApplicationController
+  def index
+    @todo = Todo.new
+  end
+end
+```
+
+If you started your server with `Rails s`, restart it with `bin/dev` to see the changes on `http://localhost:3000/todos`.
```

Comment:
I had to use `rails s` again because `bin/dev` doesn't work. Also, I do not believe the css class does anything because of my earlier issues. I am also not seeing the submit button. When I view source on the web page, the `<div class="inline">` is there, but the `<div>` is empty.